src/Symfony/Component/Serializer/Mapping/ClassMetadataInterface.php

```diff
@@ -41,7 +41,7 @@ public function addAttributeMetadata(AttributeMetadataInterface $attributeMetada
     /**
      * Gets the list of {@link AttributeMetadataInterface}.
      *
-     * @return AttributeMetadataInterface[]
+     * @return array An array of {@see AttributeMetadataInterface} instances.
```

Comment:
Keys are string (the attribute name). Static analysis tools including Scrutinizr use the following convention: `[]` for packed arrays and simply `array` for associative arrays.

Here using `[]` prevent such tools of working properly.